src/_main_/utils/utils.py

```diff
@@ -154,3 +156,6 @@ def wrapper(*args, **kwargs):
         thread = threading.Thread(target=func, args=args, kwargs=kwargs)
         thread.start()
     return wrapper
+
+def to_third_party_lang_code(language_code: str) -> str:
+    return language_code.split("-")[ 0 ] if language_code not in LANGUAGE_CODES_TO_NOT_SPLIT else language_code
```

Comment:
you probably want to check if the split yielded something of length >= 1 before you do [0]
```suggestion
    if language_code in LANGUAGE_CODES_TO_NOT_SPLIT:
        return language_code
        
    res =  language_code.split("-")
    if res:
      return res[ 0 ] 
      
     #TODO: maybe check language_code is in the list of supported languages?
     
     raise Exception("invalid language code")
```